style/js/shield.js

```diff
@@ -0,0 +1,357 @@
+
+  var renderHeight = desiredHeight / scaleHeight;
+
+  var vBaselineOffset = (desiredRenderHeight - renderHeight) / 2;
+
+  ctx.scale(scaleWidth, scaleHeight);
+  ctx.shadowColor = "white";
+  ctx.shadowBlur = 10;
+  ctx.fillText(ref, bannerIndex * 40, -1.8 * vBaselineOffset, 80);
+
+  ctx.scale(1 / scaleWidth, 1 / scaleHeight);
+}
+
+let bannerSizeH = 40;
+
+function drawBanners(c, banners) {
+  var bannerHeight = banners.length * bannerSizeH;
+  var canvas = document.createElement("canvas");
+  canvas.width = c.width;
+  canvas.height = c.height + bannerHeight;
+
+  var ctx = canvas.getContext("2d");
+  ctx.drawImage(c, 0, bannerHeight);
+
+  ctx.fillStyle = "black";
+  for (var i = 0; i < banners.length; i++) {
+    drawBannerText(c, ctx, banners[i], i);
+  }
+
+  return canvas;
+}
+
+function drawRasterShields(c, ctx, network, ref) {
+  var shieldDef = shields[network];
+  var shield;
+  var textLayout;
+
+  if (Array.isArray(shieldDef.backgroundImage)) {
+    for (var i = 0; i < shieldDef.backgroundImage.length; i++) {
+      shield = shieldDef.backgroundImage[i];
+      c.width = shield.data.width * textUpscale;
+      c.height = shield.data.height * textUpscale;
+      textLayout = layoutShieldText(c, ctx, ref, shieldDef.padding);
+      if (textLayout.fontPx > fontSizeThreshold) {
+        break;
+      }
+    }
+  } else {
+    shield = shieldDef.backgroundImage;
+    c.width = shield.data.width * textUpscale;
+    c.height = shield.data.height * textUpscale;
+    textLayout = layoutShieldText(c, ctx, ref, shieldDef.padding);
+  }
+
+  //Special cases
+  if (ref.length == 0) {
+    if (network == "US:PA:Turnpike") {
+      shield = shieldImages.shield40_us_pa_turnpike_noref;
+      c.width = shield.data.width * textUpscale;
+      c.height = shield.data.height * textUpscale;
+    } else {
+      return false;
+    }
+  }
+
+  loadShield(ctx, shield);
+
+  if (shieldDef.notext != true) {
+    ctx.fillStyle = shieldDef.textColor;
+    drawShieldText(ctx, ref, textLayout);
+  }
+  /*
+  if (typeof shieldDef.modifiers !== "undefined") {
+    c = drawBanners(c, shieldDef.modifiers);
+  }
+  */
+  return true;
+}
+
+function drawShieldsToCanvas(c, ctx, network, ref) {
+  //Default draw size
+  c.width = 80;
+  c.height = 80;
+
+  switch (network) {
+    case "US:PA:Belt":
+      ctx.fillStyle = "white";
+      ctx.fillRect(0, 0, 80, 80);
+      ctx.lineWidth = 8;
+      ctx.strokeStyle = "black";
+      ctx.strokeRect(0, 0, 80, 80);
+
+      ctx.beginPath();
+      ctx.arc(40, 40, 22, 0, 2 * Math.PI, false);
+
+      switch (ref) {
+        case "Red Belt":
+          ctx.fillStyle = "red";
+          break;
+        case "Orange Belt":
+          ctx.fillStyle = "orange";
+          break;
+        case "Yellow Belt":
+          ctx.fillStyle = "yellow";
+          break;
+        case "Green Belt":
+          ctx.fillStyle = "green";
+          break;
+        case "Blue Belt":
+          ctx.fillStyle = "blue";
+          break;
```

Comment:
Colors from Wikimedia:

```suggestion
        case "Red Belt":
          ctx.fillStyle = "#b01c2e";
          break;
        case "Orange Belt":
          ctx.fillStyle = "#d97300";
          break;
        case "Yellow Belt":
          ctx.fillStyle = "#f7d117";
          break;
        case "Green Belt":
          ctx.fillStyle = "#006b54";
          break;
        case "Blue Belt":
          ctx.fillStyle = "#003882";
          break;
```